sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1057,6 +1057,30 @@ extended either for the lifetime of the graph (including both modifiable graphs
 and the executable graphs created from them) or until the buffer is no longer
 required by the graph (such as after being replaced through executable graph update).
 
+=== Buffer Limtations
+
+Because of the delayed execution of a recorded graph it is not possible to support
+code which relies on the copy-back on destruction behaviour of buffers.
+Typically applications would rely on this behaviour to do work on the host which
+cannot inherently be captured inside a command graph. Thus when recording to a graph
+it is an error to submit a command which has an accessor on a buffer which would
+cause a write-back to happen. Using an incompatible buffer in this case will result
+in a synchronous error being thrown with error code `invalid`.
+
+The copy-back mechanism can be disabled explicitly for buffers with attached host
+storage using either `buffer::set_final_data(nullptr)` or
+`buffer::set_copy_back(false)`.
+
+When using the explicit graph API this restriction on buffers does not apply and
+the above error will not be thrown. However, because of the extension of storage
+lifetimes users should avoid relying on the copy-back mechanism in code using the
+explicit API.
```

Comment:
Would it make sense to start the section with this paragraph? Like: The following restrictions do not apply to the explicit API ... 

Advantage: if you're only interested in the explicit API you could skip the whole section.

Reply:
Yes I think that makes sense to me! I've moved the general warning against using copy-back to the above section on storage lifetimes and changed the section title to reflect that this applies only to the record and replay API.